src/wp-includes/class-wp-user.php

```diff
@@ -876,9 +899,7 @@ public function for_site( $site_id = '' ) {
 
 		$this->cap_key = $wpdb->get_blog_prefix( $this->site_id ) . 'capabilities';
 
-		$this->caps = $this->get_caps_data();
-
-		$this->get_role_caps();
+		wp_lazyload_user_meta( array( $this->ID ) );
```

Comment:
I'm not very familiar with how metadata lazy-loading works, so I feel like I'm not the right person to review this part. For example, one potential caveat here is that user metadata is generally global, but some meta keys only apply to specific sites.

I wonder for instance why this method is called here. Since user metadata is global, why would it need to be loaded (again) when the site is switched? Shouldn't this call be somewhere else, or only be called if it hadn't been called before?

Reply:
This is the key to not loading user meta on every request and is required.

Reply:
I understand that this is useful, but I don't think it's required for this PR. This line fits perfectly into the scope of https://core.trac.wordpress.org/ticket/63021, but less into lazy-loading the role/cap properties of `WP_User`.

On another note, if we keep it, could you review my previous question please?
> Since user metadata is global, why would it need to be loaded (again) when the site is switched? Shouldn't this call be somewhere else, or only be called if it hadn't been called before?

Reply:
@spacedmonkey Can you please get back to me on this open question?